backend/typescript/services/implementations/campService.ts

```diff
@@ -791,6 +801,117 @@ class CampService implements ICampService {
     }
   }
 
+  async updateCampSessionsByIds(
+    campId: string,
+    updatedCampSessions: Array<UpdateCampSessionsDTO>,
+  ): Promise<Array<CampSessionDTO>> {
+    let dbSession = null;
+    try {
+      const campSessionIds = updatedCampSessions.map((session) => session.id);
+      dbSession = await mongoose.startSession();
+      dbSession?.startTransaction();
+
+      const oldCampSessions = await MgCampSession.find({
+        _id: { $in: campSessionIds },
+        camp: { _id: campId },
+      });
+
+      if (oldCampSessions.length !== campSessionIds.length) {
+        throw new Error(
+          `Failed to update camp sessions. Could not find all camp sessions in ${campSessionIds} belonging to camp with id ${campId}`,
+        );
+      }
+
+      const camp = await MgCamp.findById(campId);
+      if (!camp) {
+        throw new Error(`Could not find camp with id=${campId}`);
+      }
+
+      const newCampSessions = await Promise.all(
+        updatedCampSessions.map(async (session) => {
+          if (camp.active) {
+            const oldCampSession: CampSession | null = await MgCampSession.findById(
+              session.id,
+            );
+            if (oldCampSession) {
+              if (
+                session.capacity &&
+                oldCampSession.campers &&
+                session.capacity < oldCampSession.campers.length
+              ) {
+                throw new Error(
+                  `Cannot decrease capacity to current number of registered campers. Requested capacity change: ${session.capacity}, current number of registed campers: ${oldCampSession.campers.length}`,
```

Comment:
shouldn't this be "cannot decrease capacity to less than current number of registered campers"?